src/lib/utils/fetch-cost-of-carry.ts

```diff
@@ -40,9 +48,87 @@ export async function fetchCostOfCarry(
       return
     }
 
-    setCostOfCarry(
-      Number(borrowedAsset.supplyAPY) - Number(borrowedAsset.variableBorrowAPY),
+    const collateralDebtTokens =
+      leverageCollateralDebt[inputOutputToken.arbitrumAddress!]
+
+    const collateralAmountPromise = publicClient.readContract({
+      address: collateralDebtTokens.collateralAmountToken as Address,
+      abi: ArbitrumLeverageTokenAbi,
+      functionName: 'balanceOf',
+      args: [inputOutputToken.arbitrumAddress as Address],
+    })
+    const collateralDecimalsPromise = publicClient.readContract({
+      address: collateralDebtTokens.collateralAmountToken as Address,
+      abi: ArbitrumLeverageTokenAbi,
+      functionName: 'decimals',
+    })
+    const debtAmountPromise = publicClient.readContract({
+      address: collateralDebtTokens.debtAmountToken as Address,
+      abi: ArbitrumLeverageTokenAbi,
+      functionName: 'balanceOf',
+      args: [inputOutputToken.arbitrumAddress as Address],
+    })
+    const debtDecimalsPromise = publicClient.readContract({
+      address: collateralDebtTokens.debtAmountToken as Address,
+      abi: ArbitrumLeverageTokenAbi,
+      functionName: 'decimals',
+    })
+
+    const [collateralAmount, collateralDecimals, debtAmount, debtDecimals] =
+      await Promise.all([
+        collateralAmountPromise,
+        collateralDecimalsPromise,
+        debtAmountPromise,
+        debtDecimalsPromise,
+      ])
+
+    const navProvider = new NavProvider()
+    const nav = await navProvider.getNavPrice(
+      inputOutputToken.symbol,
```

Comment:
minor: would this be more clear to just call it `indexToken`?

Reply:
good call - updated 👍